### ISSUE TITLE
Implement buildkit cache mounts in the Dockerfile to speed up builds

### ISSUE THREAD

**Issue by calvinhp** [sixfeetup/scaf#19]
Docker's BuildKit provides advanced caching capabilities that can be integrated with GitHub Actions. By using the docker/build-push-action, you can leverage the GitHub Actions cache to store and retrieve Docker layers:

```
- name: Build Image
  uses: docker/build-push-action@v5
  with:
    context: .
    file: ./Dockerfile
    tags: ${{ env.IMAGE_TAG }}
    cache-from: type=gha
    cache-to: type=gha,mode=max

```
This setup allows you to cache layers between builds, reducing build times significantly after the initial build.

**Comment by sedatbasar** [comment]
Blocked by: #119

**Comment by gfranxman** [comment]
on hold in case this has to be adjusted after #73 is complete.
